library/src/main/java/org/apache/apex/malhar/lib/fs/redshift/RedshiftOutputModule.java

```diff
@@ -0,0 +1,379 @@
+    } else {
+      FSRecordCompactionOperator<byte[]> hdfsWriteOperator = dag.addOperator("WriteToHDFS", new FSRecordCompactionOperator<byte[]>());
+      hdfsWriteOperator.setConverter(new GenericFileOutputOperator.NoOpConverter());
+      if (maxLengthOfRollingFile != null) {
+        hdfsWriteOperator.setMaxLength(maxLengthOfRollingFile);
+      }
+      input.set(hdfsWriteOperator.input);
+
+      RedshiftJdbcTransactionableOutputOperator redshiftOutput = dag.addOperator("LoadToRedshift", createRedshiftOperator());
+
+      dag.addStream("load-to-redshift", hdfsWriteOperator.output, redshiftOutput.input);
+    }
+  }
+
+  /**
+   * Create the RedshiftJdbcTransactionableOutputOperator instance
+   * @return RedshiftJdbcTransactionableOutputOperator object
+   */
+  protected RedshiftJdbcTransactionableOutputOperator createRedshiftOperator()
+  {
+    RedshiftJdbcTransactionableOutputOperator redshiftOutput = new RedshiftJdbcTransactionableOutputOperator();
+    redshiftOutput.setAccessKey(accessKey);
+    redshiftOutput.setSecretKey(secretKey);
+    if (bucketName != null) {
+      redshiftOutput.setBucketName(bucketName);
+    }
+    redshiftOutput.setTableName(tableName);
+    if (emrClusterId != null) {
+      redshiftOutput.setEmrClusterId(emrClusterId);
+    }
+    redshiftOutput.setReadFromS3(readFromS3);
+    redshiftOutput.setStore(store);
+    redshiftOutput.setBatchSize(batchSize);
+    redshiftOutput.setRedshiftDelimiter(redshiftDelimiter);
+    redshiftOutput.setRegion(region);
+    return redshiftOutput;
+  }
+
+  /**
+   * Generic implementation for S3TupleOutputModule.S3BytesOutputModule and the s3Reconciler is of type S3CompactionOperator.
+   */
+  public static class TupleBasedS3 extends S3TupleOutputModule.S3BytesOutputModule
+  {
+    public final transient ProxyOutputPort<FSRecordCompactionOperator.OutputMetaData> output = new ProxyOutputPort<>();
+    @Override
+    public S3Reconciler createReconciler()
+    {
+      return new S3CompactionOperator();
+    }
+
+    @Override
+    public void populateDAG(DAG dag, Configuration conf)
+    {
+      super.populateDAG(dag, conf);
+      if (getS3Reconciler() instanceof S3CompactionOperator) {
+        output.set(((S3CompactionOperator)getS3Reconciler()).outputPort);
+      }
+    }
+  }
+
+  /**
+   * This operator uploads files to Amazon S3 after files are finalized and
+   * frozen by the committed callback. This operator emits the OutputMetaData which are uploaded to S3.
+   */
+  public static class S3CompactionOperator extends S3Reconciler
+  {
+    public final transient DefaultOutputPort<FSRecordCompactionOperator.OutputMetaData> outputPort = new DefaultOutputPort<>();
+
+    @Override
+    protected void removeIntermediateFiles(FSRecordCompactionOperator.OutputMetaData metaData)
+    {
+      super.removeIntermediateFiles(metaData);
+      metaData.setPath(getDirectoryName() + Path.SEPARATOR + metaData.getFileName());
```

Comment:
Added more explanation under javadocs. Adding the same behaviour at parent level is not feasible as this behaviour is needed for redshift.